apps/api/src/controllers/post.ts

```diff
@@ -0,0 +1,103 @@
+import { RequestHandler } from 'express';
+import Post from '../models/post';
+import Comment from '../models/comment';
+
+/**
+ Get all posts 
+ */
+export const getPosts: RequestHandler = async (_req, res) => {
+  try {
+    const posts = await Post.find();
```

Comment:
```suggestion
    const posts = await Post.find().populate('category').exec();
```

With the change in the model here you can populate the data of the category to receive the category id and the category name, same for get post by id and get posts by category with this change all the tests will pass